python/pyspark/sql/dataframe.py

```diff
@@ -146,9 +146,74 @@ def insertInto(self, tableName, overwrite=False):
         """
         self._jdf.insertInto(tableName, overwrite)
 
-    def saveAsTable(self, tableName):
-        """Creates a new table with the contents of this DataFrame."""
-        self._jdf.saveAsTable(tableName)
+    def _java_save_mode(self, mode):
+        """Returns the Java save mode based on the Python save mode represented by a string.
+        """
+        jmode = self._sc._jvm.org.apache.spark.sql.sources.SaveMode.ErrorIfExists
+        mode = mode.lower()
+        if mode == "append":
+            jmode = self._sc._jvm.org.apache.spark.sql.sources.SaveMode.Append
+        elif mode == "overwrite":
+            jmode = self._sc._jvm.org.apache.spark.sql.sources.SaveMode.Overwrite
+        elif mode == "ignore":
+            jmode = self._sc._jvm.org.apache.spark.sql.sources.SaveMode.Ignore
+        elif mode == "error":
+            pass
+        else:
+            raise ValueError(
+                "Only 'append', 'overwrite', 'ignore', and 'error' are acceptable save mode.")
+        return jmode
+
+    def saveAsTable(self, tableName, source=None, mode="append", **options):
```

Comment:
I saw it's only used int insertInto() and saveAsTable(), and registerAsTable() use `name`.

It's not a keyword argument, so it's safe to change the name.